File: bms/go/go-gt/delta/delta.go
package test

var ()

const ()

func delta(a *Matrix, b *Matrix, p Vector, r int64, s int64) (d int64) {
	var i int64
	d = int64((a.Get(r, r)-a.Get(s, s))*(b.Get(p[s], p[s])-b.Get(p[r], p[r])) +
		(a.Get(r, s)-a.Get(s, r))*(b.Get(p[s], p[r])-b.Get(p[r], p[s])))
	for i = 0; i < p.Len(); i++ {
		if i != r && i != s {
			d += (a.Get(i, r)-a.Get(i, s))*(b.Get(p[i], p[s])-b.Get(p[i], p[r])) +
				(a.Get(r, i)-a.Get(s, i))*(b.Get(p[s], p[i])-b.Get(p[r], p[i]))
		}
	}
	return d
}

func (m Matrix) Get(i int64, j int64) int64 {
	return m.A[i*m.N+j]
}

func (v Vector) Len() int64 {
	return int64(len(v))
}

type Vector []int64

type Matrix struct {
	N int64
	A []int64
}

File: translations_only/success/go/go-gt/delta/claude2/Restart/delta.rs

use std::convert::TryInto;

struct Matrix {
    n: i64,
    a: Vec<i64>, 
}

struct Vector(Vec<usize>);

fn get(m: &Matrix, i: usize, j: usize) -> i64 {
    m.a[i * m.n as usize + j]
}

fn delta(a: &Matrix, b: &Matrix, p: &Vector, r: usize, s: usize) -> i64 {
    let mut d = (get(a, r, r) - get(a, s, s)) * (get(b, p.0[s].try_into().unwrap(), p.0[s].try_into().unwrap()) - get(b, p.0[r].try_into().unwrap(), p.0[r].try_into().unwrap()))
        + (get(a, r, s) - get(a, s, r)) * (get(b, p.0[s].try_into().unwrap(), p.0[r].try_into().unwrap()) - get(b, p.0[r].try_into().unwrap(), p.0[s].try_into().unwrap()));

    let len = p.0.len();
    let mut i = 0;
    while i < len {
        if i != r && i != s {
            let tmp = (get(a, i, r) - get(a, i, s)) * (get(b, p.0[i].try_into().unwrap(), p.0[s].try_into().unwrap()) - get(b, p.0[i].try_into().unwrap(), p.0[r].try_into().unwrap()))
                + (get(a, r, i) - get(a, s, i)) * (get(b, p.0[s].try_into().unwrap(), p.0[i].try_into().unwrap()) - get(b, p.0[r].try_into().unwrap(), p.0[i].try_into().unwrap()));
            d += tmp;
        }
        i += 1;
    }

    d
}

